sycl/include/sycl/ext/intel/esimd/detail/operators.hpp

```diff
@@ -548,17 +548,88 @@ __ESIMD_DEF_SIMD_VIEW_CMP_OP(>=, __ESIMD_DNS::is_simd_type_v<SimdT1>)
+    SimdTx res(0);                                                             \
+    res.merge(RHS, LHS);                                                       \
+    return res;                                                                \
+  }                                                                            \
+                                                                               \
+  /* SCALAR * simd_mask  */                                                    \
+  template <class T1, class SimdT2, class T2 = typename SimdT2::element_type,  \
+            class SimdT1 = typename __ESIMD_NS::simd<T1, SimdT2::length>,      \
+            class Tx = __ESIMD_DNS::computation_type_t<T1, T2>,                \
+            class SimdTx = __ESIMD_NS::simd<Tx, SimdT2::length>>               \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT2> && COND,    \
+                          SimdTx>                                              \
+  operator*(T1 LHS, const SimdT2 &RHS) {                                       \
+    SimdTx res(0);                                                             \
+    res.merge(LHS, RHS);                                                       \
+    return res;                                                                \
+  }                                                                            \
+                                                                               \
+  /* simd_mask * SCALAR */                                                     \
+  template <class SimdT1, class T2, class T1 = typename SimdT1::element_type,  \
+            class SimdT2 = typename __ESIMD_NS::simd<T2, SimdT1::length>,      \
+            class Tx = __ESIMD_DNS::computation_type_t<T1, T2>,                \
+            class SimdTx = __ESIMD_NS::simd<Tx, SimdT1::length>>               \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT1> && COND,    \
+                          SimdTx>                                              \
+  operator*(const SimdT1 &LHS, T2 RHS) {                                       \
+    SimdTx res(0);                                                             \
+    res.merge(RHS, LHS);                                                       \
+    return res;                                                                \
+  }                                                                            \
+  /* simd_mask * simd_mask  */                                                 \
+  template <class SimdT, class T1 = typename SimdT::element_type,              \
+            class T2 = typename SimdT::element_type>                           \
+  inline std::enable_if_t<__ESIMD_DNS::is_simd_mask_type_v<SimdT> && COND,     \
+                          __ESIMD_NS::simd<T2, SimdT::length>>                 \
+  operator*(const SimdT &LHS, const SimdT &RHS) {                              \
```

Comment:
Personally, I like this variant more: simd_mask * simd_mask gives simd_mask.

If choose to be consistent with CM language, and we do that. Then simd_mask<N> is simply simd<uint16_t, N>, and thus the C++ type promotion rules are used and the result is simd<uint32_t, N>.

Same comment for other ops like BINOP